packages/@o3r/design/src/core/design-token/parsers/design-token.parser.ts

```diff
@@ -0,0 +1,172 @@
+import { promises as fs } from 'node:fs';
+import type { DesignTokenVariableSet, DesignTokenVariableStructure } from './design-token-parser.interface';
+import type {
+  DesignToken,
+  DesignTokenExtensions,
+  DesignTokenGroup,
+  DesignTokenGroupExtensions,
+  DesignTokenNode,
+  DesignTokenSpecification
+} from '../design-token-specification.interface';
+import {
+  DesignTokenTypeStrokeStyleValue,
+  isDesignToken,
+  isDesignTokenGroup,
+  isTokenTypeStrokeStyleValueComplex
+} from '../design-token-specification.interface';
+
+interface ParentReference {
+  name: string;
+  tokenNode: DesignTokenGroup;
+}
+
+const tokenReferenceRegExp = /\{([^}]+)\}/g;
+
+const getTokenReferenceName = (tokenName: string, parents: string[]) => (`${parents.join('.')}.${tokenName}`);
+const getCssVarName = (tokenName: string, parents: string[]) => (getTokenReferenceName(tokenName, parents).replace(/[ .]+/g, '-'));
+const getGroups = (tokenName: string, parents: string[]) => (getTokenReferenceName(tokenName, parents).split(/[ .]+/).slice(0, -1));
+const getExtensions = (parentNode: DesignTokenNode[]) => parentNode.reduce((acc, node) => ({...acc, ...node.$extensions}), {} as DesignTokenGroupExtensions & DesignTokenExtensions);
+const getReferences = (cssRawValue: string) => Array.from(cssRawValue.matchAll(tokenReferenceRegExp)).map(([,tokenRef]) => tokenRef);
+
+// eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
```

Comment:
Why do you need to disable eslint here ?

Reply:
Because of the `| string` that override a part of the litteral types from `DesignTokenVariableSet`.
(The `| string` is mandatory to handle the variable reference)